scripts/dss-start-stop--think-before-you-invoke-me.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+"""
+Use the AWS Lambda concurrency property to disable/enable lambda functions:
+    - set concurrency to 0 to disable execution
+    - remove concurrency setting to enable execution
+"""
+import os
+import sys
+import boto3
+import click
+import argparse
+
+
+stage = os.environ['DSS_DEPLOYMENT_STAGE']
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("--stop", action="store_true")
+parser.add_argument("--start", action="store_true")
+args = parser.parse_args()
+assert(
+    (args.start or args.stop) and not (args.start and args.stop)
+)
+
+
+action = f"DSS {stage} START" if args.start else f"DSS {stage} STOP"
+if not click.confirm(f"Are you sure you want to do this ({action})?"):
+    sys.exit(0)
+
+
+LAMBDA = boto3.client('lambda')
+
+
+def disable_lambda(name):
+    LAMBDA.put_function_concurrency(
+        FunctionName=name,
+        ReservedConcurrentExecutions=0
+    )
+    print(f"halted {name}")
+    
+def enable_lambda(name):
+    LAMBDA.delete_function_concurrency(
+        FunctionName=name,
+    )
+    print(f"started {name}")
+
+
+for root, dirs, files in os.walk(os.path.join(pkg_root, 'daemons')):
+    functions = [f'{name}-{stage}' for name in dirs]
```

Comment:
Don't you need to concat `dir` and `root`? Make sure this doesn't just work coincidentally because you're running the script from the project root.

You could eliminate the odd loop and break construct with 

```
root, dirs, files = next(os.walk(os.path.join(pkg_root, 'daemons')))
```

That would raise if `./daemons` were empty and I'd actually prefer that.

Reply:
The concatenation is not necessary, it seems.

Changed to the `next()` construct.